manifests/config.pp

```diff
@@ -179,20 +179,27 @@
       }
     }
     'systemd': {
-      systemd::unit_file {'prometheus.service':
+      file { '/etc/systemd/system/prometheus.service':
+        mode    => '0644',
+        owner   => 'root',
+        group   => 'root',
+        notify  => $notify,
         content => epp("${module_name}/prometheus.systemd.epp", {
           'user'           => $prometheus::server::user,
           'group'          => $prometheus::server::group,
           'daemon_flags'   => $daemon_flags,
           'max_open_files' => $max_open_files,
           'bin_dir'        => $prometheus::server::bin_dir,
         }),
-        notify  => $notify,
+      }
+      exec { 'prometheus-systemd-reload':
+        command     => 'systemctl daemon-reload',
+        path        => [ '/usr/bin', '/bin', '/usr/sbin' ],
+        refreshonly => true,
```

Comment:
You don't notify this resource so you are introducing a bug here. That's sort of the point of using a module to do so: you abstract away resources so you don't repeat yourself and introduce bugs.

Once we drop support for Puppet < 6.1.0 (which will be in November), then I'd agree there's not really any value.